### PR TITLE
 Add ban/kick concommand completion

### PR DESCRIPTION
**Pull request by catornot**
just adds completion to ban, kick and unban commands.

image

## how test this ??????

try banning, kicking and unbanning someone from the in game console, it should display the closest uid/name to what you typed in :)

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
> huh build failed to configure cmake?

`minizip` apparently has a remote dependency on https://sourceware.org/ which was done at the time xD

File: NorthstarDLL/server/auth/bansystem.cpp [hunk 1]
```diff
@@ -213,12 +216,59 @@ void ConCommand_clearbanlist(const CCommand& args)
...
-	RegisterConCommand("ban", ConCommand_ban, "bans a given player by uid or name", FCVAR_GAMEDLL);
-	RegisterConCommand("unban", ConCommand_unban, "unbans a given player by uid", FCVAR_GAMEDLL);
+	RegisterConCommand("ban", ConCommand_ban, "bans a given player by uid or name", FCVAR_GAMEDLL, ConCommand_banCompletion);
+	RegisterConCommand("unban", ConCommand_unban, "unbans a given player by uid", FCVAR_GAMEDLL, ConCommand_banCompletion);
 	RegisterConCommand("clearbanlist", ConCommand_clearbanlist, "clears all uids on the banlist", FCVAR_GAMEDLL);
 }
+
+ON_DLL_LOAD_RELIESON("server.dll", KickCompletion, ConCommand, (CModule module))
+{
+	ConCommand* kick = R2::g_pCVar->FindCommand("kick");
```
Comment (GeckoEidechse) [inline NorthstarDLL/server/auth/bansystem.cpp:271]: @catornot builds are failing atm as `R2` namespace was removed

Comment (catornot) [reply NorthstarDLL/server/auth/bansystem.cpp]: removed it

**Review by ASpoonPlaysGames — APPROVED** [review]
Code looks aight to me, would be cool to figure out what those flags are.

Would also be cool if `unban` would parse banlist or something, since going through the connected players for `unban` command completion is a bit dumb...

File: primedev/server/auth/bansystem.cpp [hunk 2]
```diff
@@ -213,12 +216,59 @@ void ConCommand_clearbanlist(const CCommand& args)
...
+
 ON_DLL_LOAD_RELIESON("engine.dll", BanSystem, ConCommand, (CModule module))
 {
 	g_pBanSystem = new ServerBanSystem;
 	g_pBanSystem->OpenBanlist();
 
-	RegisterConCommand("ban", ConCommand_ban, "bans a given player by uid or name", FCVAR_GAMEDLL);
-	RegisterConCommand("unban", ConCommand_unban, "unbans a given player by uid", FCVAR_GAMEDLL);
+	RegisterConCommand("ban", ConCommand_ban, "bans a given player by uid or name", FCVAR_GAMEDLL, ConCommand_banCompletion);
+	RegisterConCommand("unban", ConCommand_unban, "unbans a given player by uid", FCVAR_GAMEDLL, ConCommand_banCompletion);
```
Comment (GeckoEidechse) [inline primedev/server/auth/bansystem.cpp:265]: Not sure if it makes sense for `unban` to offer to complete from players currently on the server as naturally those are not banned. Instead if offering completion, it should be from the list of players that are in banlist. But naturally that's outside the scope of this PR.

For this PR I'd suggest to just drop adding it to `unban` ^^

Comment (catornot) [reply primedev/server/auth/bansystem.cpp:265]: Yeah, that would make more sense 😅

Comment (GeckoEidechse) [reply primedev/server/auth/bansystem.cpp]: Adressed in a40fef643abc8e69717677594c8976285e12317f

**Review by GeckoEidechse — APPROVED** [review]
Kinda forgot to mention that I already tested this PR and it worked as expected ^^

**Pull request merged**